packages/mui-infra/package.json

```
@@ -0,0 +1,74 @@
+{
+  "name": "@mui/infra",
```

@oliviertassinari What do you recommend the package name to be ?
`@mui/internal-infra` or `@mui-internal/infra` or does the current name work?

---

What I see in https://github.com/mui/base-ui/pull/2008/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R76 looks correct for the prefixes, it matches with https://www.notion.so/mui-org/engineering-mui-utils-purpose-9a9fc9da3a004864b6c4e1f4d1f24f95?source=copy_link#d1699bb48f0f4ef58ca831d1ee9772ec. 

For the actual name of this, I would always be more in favor of having ONE big npm package, than many, per https://www.notion.so/mui-org/Engineering-philosophy-ec1615511f594782a8a2adeec3e25143?source=copy_link#1e7cbfe7b660805eb1f3f2475839085f so we are forced to update everything at once, so we are forced to make small breaking changes and so we don't have to deal with different version intern compatibility. So something like: 

```suggestion
  "name": "@mui/internal-code-infra",
```

sounds amazing.

---

1. Agree
2. This is tricky, it's possible with pnpm linking but it can get messy with transitive dependencies and singletons. Otherwise use `file:` [dependencies](https://pnpm.io/cli/link#whats-the-difference-between-pnpm-link-and-using-the-file-protocol). but you'll have to re-install on every change. There is no having your cake and eating it too. Going multi-repo **will** impose degraded DX on shared infra, always.
3. Again tricky, everything with trade-offs:
  - `github:` dependecy: no build step posseble, no `workspace:` dependencies possible. slow install process
  - renovatebot +
    - rely on pkg.pr.new: much lower guarantees on stability than npm. If it goes down for a few days, we are in trouble. This is not imaginary, we've had this situation in smaller scale with csbci and Base UI.
    - publish every master commit to npm: tricky versioning situation, version number in github bisects from version in npm
      - sequential version: great for natural ordering, npm can take two versions and compare them know which one is newer
      - versioning by SHA, rely on npm tag. no ordering at all in version numbers, npm will just install the alphabetically last version, even if it's not the last published one. Only meaningful with npm publish tags. And only one version will be relevant, i.e. the one that is tagged

    best trade-off imo is to automatically publish to npm on every push to master, and use a sequential patch version. I suppose in source we could set the version in package.json to `1.2.x`, the publish flow can then determine the patch by looking at latest published version that satisfies `1.2.x` and increment. This way we can still control minor and major updates. renovatebot can then just operate on regular semver, which should be quite controllable as well.